File: 0011-container-with-most-water/0011-container-with-most-water.c
int maxArea(int* height, int heightSize) {
        int i = 0;
    int j = heightSize - 1;
    int result = 0;
    int temp = 0;
    while (j > i) {
        int h1 = height[i];
        int h2 = height[j];
        temp = (j - i) * ((h1 < h2) ? h1 : h2);
        if (temp > result) result = temp;
        if (h1 >= h2) {
            j--;
        } else {
            i++;
        }
    }
    return result;

}

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left =0,right = height.length-1;
        int maxarea = Integer.MIN_VALUE;
        while(left<=right){
            int mid = Math.min(height[left],height[right]);
            int wid =  right - left;

            maxarea = Math.max(maxarea,mid*wid);

            if(height[left] < height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return maxarea;
    }
}